python_modules/dagster/dagster/_grpc/client.py

```diff
@@ -252,10 +256,31 @@ def external_repository(self, external_repository_origin):
             serialized_repository_python_origin=serialize_dagster_namedtuple(
                 external_repository_origin
             ),
+            defer_snapshots=defer_snapshots,
         )
 
         return res.serialized_external_repository_data
 
+    def external_job(
+        self,
+        external_repository_origin: ExternalRepositoryOrigin,
+        job_name: str,
+    ) -> str:
```

Comment:
what's the back-compat plan here in OSS - to make this feature opt-in and if you opt-in you need to make sure everything is upgraded? or do so some kind of fallback in the future if we add it?

Reply:
This grpc method is only called when the deferred snapshots responses comes back, so we only call it when we we know we are hitting a recent-enough grpc server. 

If an external_job call got added somewhere else that could be a problem.